openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/Openc3Screen.vue

```diff
@@ -738,11 +738,11 @@ export default {
           ) {
             widget.settings.push(setting.slice(1))
           }
-          // Recursively apply to all widgets contained in layouts
-          if (widget.widgets) {
-            this.applyGlobalSettings(widget.widgets)
-          }
         })
+        // Recursively apply to all widgets contained in layouts
+        if (widget.widgets) {
+          this.applyGlobalSettings(widget.widgets)
+        }
```

Comment:
Nesting this inside the forEach loops caused the page to freeze if you have more than a few global settings and nested widgets.